Fixed handling of nan for evenly_distribute_backward

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45280 Fixed handling of nan for evenly_distribute_backward**

Performance is the same on CPU and on CUDA is only 1-1.05x slower. This change is necessary for the future nan ops including nan(min|max|median)

Differential Revision: [D23908796](https://our.internmc.facebook.com/intern/diff/D23908796)

heitorschueroff committed:
Corrected handling of nan for evenly_distribute_backward

[ghstack-poisoned]

albanD reviewed (commented):
Looks good!

Maybe add a test to make sure it has the right behavior now both on CPU and CUDA?

heitorschueroff commented:
> Looks good!
> 
> Maybe add a test to make sure it has the right behavior now both on CPU and CUDA?

I'll add some test now but more will come on the PR for the nan ops.

heitorschueroff committed:
Update on "Fixed handling of nan for evenly_distribute_backward"


Performance is the same on CPU and on CUDA is only 1-1.05x slower. This change is necessary for the future nan ops including nan(min|max|median)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Fixed handling of nan for evenly_distribute_backward"


Performance is the same on CPU and on CUDA is only 1-1.05x slower. This change is necessary for the future nan ops including nan(min|max|median)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM!

heitorschueroff committed:
Update on "Fixed handling of nan for evenly_distribute_backward"


Performance is the same on CPU and on CUDA is only 1-1.05x slower. This change is necessary for the future nan ops including nan(min|max|median)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Fixed handling of nan for evenly_distribute_backward"


Performance is the same on CPU and on CUDA is only 1-1.05x slower. This change is necessary for the future nan ops including nan(min|max|median)

Differential Revision: [D23908796](https://our.internmc.facebook.com/intern/diff/D23908796)

[ghstack-poisoned]
